### ISSUE TITLE
Highlights 'off' in some searches

### ISSUE THREAD

**Issue by DanilaFe** [jameslittle230/stork#94]
Hi! Congrats on the recent 1.0 release. Stork is looking pretty good!

I'm running into a strange highlight issue while integrating stork into my site. I'm able to get it to load the index and search, but when I get results, the `stork-highlight` elements seem misplaced. Here's what it looks like:

![Screenshot_2020-12-27 Daniel's Blog(4)](https://user-images.githubusercontent.com/4361282/103189311-eaf6f900-4880-11eb-8547-958f6c2157d4.png)

Double checking, the issue is in HTML. Here's one of the entries:
```HTML
<p class="stork-excerpt">
        ...the user. Also, when there’s no error, our
        co<span class="stork-highlight">mpiler d</span>oesn’t really tell us anything at all...
</p>
```

I'm not quite sure what's going on. From what I know, Stork should _only_ be using the index, both for searching and displaying results, so the issue doesn't seem to be on my end. My index file is attached, in case it helps: [index.st.zip](https://github.com/jameslittle230/stork/files/5745850/index.st.zip)

Thanks for making this! It's a breeze to set up and use.

**Comment by jameslittle230** [comment]
@DanilaFe -

I've tracked down the bug -- I was counting bytes, not unicode characters, to derive the highlight offset. Your content uses `’` (`U+2019`, AKA `RIGHT SINGLE QUOTATION MARK`), which happens to take up two bytes if you count its length in Rust.

I've (written a test for this and) switched to a UTF-8-safe character counter. I can then verify with your index that it's highlighting `Compiler` correctly:

<img width="564" alt="Screen Shot 2020-12-28 at 6 38 12 PM" src="https://user-images.githubusercontent.com/3058200/103249142-d9265c00-493b-11eb-800e-fe345823d494.png">

This will get fixed in 1.0.2, and won't require an index rebuild (since it was a bug in the WASM, not in the indexer).

Thanks for reporting this! Much appreciated.

James

**Comment by DanilaFe** [comment]
Seems to work with the latest version. Thank you!
